mail_show_follower/models/mail_mail.py

```diff
@@ -60,96 +60,34 @@ def remove_p(markup_txt):
-                            )
-                        if cc_internal:
-                            partners = partners_obj.filtered(
-                                lambda x: x.id not in user_partner_ids
-                                and (
-                                    not x.user_ids
-                                    or any(x.mapped("user_ids.show_in_cc"))
-                                )
-                            )
-                        else:
-                            partners = partners_obj.filtered(
-                                lambda x: x.id not in user_partner_ids
-                                and (
-                                    not x.user_ids
-                                    or group_portal in x.user_ids.mapped("groups_id")
-                                )
-                            )
-                        partners = partners.filtered(
-                            lambda x: not x.user_ids
-                            or x.user_ids  # otherwise, email is not sent
-                            and "email" in x.user_ids.mapped("notification_type")
-                        )
-                        # set proper lang for recipients
-                        langs = list(
-                            filter(
-                                bool,
-                                mail.mapped("recipient_ids.lang")
-                                + [
-                                    mail.author_id.lang,
-                                    self.env.company.partner_id.lang,
-                                ],
-                            )
-                        )
-                        # get show follower text
-                        final_cc = mail.with_context(
-                            lang=langs and langs[0]
-                        )._build_cc_text(partners)
-                        # it is saved in the body_html field so that it does
-                        # not appear in the odoo log
-                        mail.body_html = final_cc + mail.body_html
+
+        for mail in self:
+            if not (mail.model and mail.res_id and group_portal):
+                continue
+            recipients = mail.recipient_ids | mail.notified_partner_ids
+            record = self.env[mail.model].browse(mail.res_id)
+            company = (
+                record.company_id
+                if hasattr(record, "company_id")
+                else self.env.user.company_id
+            )
+            show_internal_users = company and company.show_internal_users_cc
+            show_in_cc_recipients = recipients.filtered(
+                lambda r: r._show_in_cc(show_internal_users)
+            )
+            if len(show_in_cc_recipients) <= 1:
+                continue
+
+            langs = (
+                mail.recipient_ids.mapped("lang")
+                or mail.author_id.lang
+                or self.env.company.partner_id.lang
```

Comment:
issue: better reuse the company from the record if available

```suggestion
                or company.partner_id.lang
```